feat: ✨ create summary page

KevinWu098 opened the pull request:
## Summary
- Horizontal scrollable cards with groups
- Vertical meeting cards in two tabs: Scheduled, Unscheduled
  - Scheduled Meetings are grouped by date and have an "attending status" radio group
  - Unscheduled Meetings are grouped by start date
- Adds unplugin-icons, as our icon set provider -- I'm using Material Symbols

Desktop:
<img width="1568" alt="Screenshot 2023-12-20 at 4 52 09 AM" src="https://github.com/icssc/ZotMeet/assets/100006999/07613c7e-b3da-4958-b7d9-62882c99432a">

Mobile:
<img width="355" alt="Screenshot 2023-12-20 at 4 56 06 AM" src="https://github.com/icssc/ZotMeet/assets/100006999/8eb3d73a-634f-49b0-b150-95311b9572bd">

## Issues
Closes #6

## Future Followup
1. Shot myself in the foot a bit with how I implemented Group cards, but background images have no separate opacity from text, so that's something I'd like to fix when we delve into Groups more
2. Alert Icons

KevinWu098 commented:
P.S. I am wondering about what icon library / options are available for SvelteKit, since I didn't see any built into Skeleton

ap0nia commented:
[unplugin-icons](https://github.com/unplugin/unplugin-icons) with the [@iconify/json](https://www.npmjs.com/package/@iconify/json) icon set if you just want access to all of the icons [on this site](https://icon-sets.iconify.design/)

KevinWu098 commented:
<img width="394" alt="Screenshot 2023-12-12 at 12 17 42 PM" src="https://github.com/icssc/ZotMeet/assets/100006999/8a307a24-9bf0-48f5-b1aa-6414f8e53ea1">

Current draft; still experimenting with doing a two-tab view for scheduled v. unscheduled, how to handle recurring meetings etc

MinhxNguyen7 commented:
> P.S. I am wondering about what icon library / options are available for SvelteKit, since I didn't see any built into Skeleton

@ap0nia's seems fine. I don't have an opinion, so do what you think is good @KevinWu098.

Also, next time, if you're looking for a reply from me, tag me. I don't just go through every PR's comments regularly lol.

KevinWu098 commented:
> > P.S. I am wondering about what icon library / options are available for SvelteKit, since I didn't see any built into Skeleton
> 
> 
> 
> @ap0nia's seems fine. I don't have an opinion, so do what you think is good @KevinWu098.
> 
> 
> 
> Also, next time, if you're looking for a reply from me, tag me. I don't just go through every PR's comments regularly lol.

Haha will do; I didn't want to tag ya during finals week, since an icon set wasn't too pressing

MinhxNguyen7 commented:
Thanks for the consideration, but just tag me 👍. If I'm busy, I'll leave it for later.

KevinWu098 commented:
<img width="476" alt="Screenshot 2023-12-20 at 2 17 54 AM" src="https://github.com/icssc/ZotMeet/assets/100006999/eb16ad94-5f24-49c8-a3ed-d045c8164ecc">

Current iteration: 
1. Added in an "attendance" button, similar to how Google Calendar/Gmail does it 
2. Added in the location as a field

In progress work:
1. Two tabs for scheduled vs unscheduled
2. Handle dead space on wider screens

<img width="1568" alt="Screenshot 2023-12-20 at 2 19 50 AM" src="https://github.com/icssc/ZotMeet/assets/100006999/ea257e25-fe24-4454-b483-e410d9ca1648">

MinhxNguyen7 commented on src/routes/summary/+page.svelte:
```diff
@@ -0,0 +1,335 @@
+        {#if tabSet === 0}
+          <div class="flex flex-col gap-4">
+            {#each Object.keys(groupedMeetingsByDate) as date}
+              <div class="p-2 card variant-ringed">
```
```suggestion
              <div class="p-2 card variant-glass">
```

MinhxNguyen7 replied:
I think that this lighter delimiter looks nicer. It's like the AntAlmanac calendar thing.

MinhxNguyen7 commented on src/routes/summary/+page.svelte:
```diff
@@ -0,0 +1,335 @@
+                  {#each groupedMeetingsByDate[date] as meeting}
+                    <a href={meeting.link} target="_blank" referrerpolicy="no-referrer">
+                      <div
+                        class="flex flex-col justify-between gap-4 p-3 bg-center bg-cover rounded-lg md:items-center h-fit md:flex-row card variant-ghost hover:variant-filled-surface"
```
Similarly, I think the highlighting on this card should be lighter as well. It doesn't take much to percieve the delimitation.

MinhxNguyen7 commented on src/lib/components/summary/GroupList.svelte:
```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
```
This should be named GroupsCarousel, and the "GROUPS" header should be part of the parent component.

MinhxNguyen7 reviewed (changes requested):
This looks great! There are just a few things to be ironed out, in addition to the comments:

- Instead of clicking on the card taking you to the meeting page, I think only the meeting name should do that. It's too easy to fat-finger right now, I'd say. Also, we might want the location to link to something in the future.
- On mobile, the meeting card stays highlighted after toggling attendance. The previous point would fix this I'd assume.
- The Scheduled/Unscheduled tabs are misplaced on desktop. Desktop is not a priority, but this should be an easy fix.
- Similarly, the groups carousel should display a small side scrollbar on desktop.

Also, for easier integration, can you create a store for the meetings and groups and inject the boilerplate from there?

All in all, good work!

KevinWu098 commented:
> This looks great! There are just a few things to be ironed out, in addition to the comments:
> 
> * Instead of clicking on the card taking you to the meeting page, I think only the meeting name should do that. It's too easy to fat-finger right now, I'd say. Also, we might want the location to link to something in the future.
> * On mobile, the meeting card stays highlighted after toggling attendance. The previous point would fix this I'd assume.
> * The Scheduled/Unscheduled tabs are misplaced on desktop. Desktop is not a priority, but this should be an easy fix.
> * Similarly, the groups carousel should display a small side scrollbar on desktop.
> 
> Also, for easier integration, can you create a store for the meetings and groups and inject the boilerplate from there?
> 
> All in all, good work!

Aye aye, will do!

MinhxNguyen7 commented:
@KevinWu098 Although I did say that we should communicate frequently, you don't need to say that you'll address review comments lol. Thanks for the enthusiasm though.

MinhxNguyen7 commented on src/routes/summary/+page.svelte:
```diff
@@ -0,0 +1,335 @@
+      </div>
+
+      <svelte:fragment slot="panel">
+        {#if tabSet === 0}
```
Can you extract this to two components, `ScheduledMeetingsList` and `UnscheduledMeetingsList`? They are relatively distinct, and this file is getting a little long. That might necessaitate the moving of the conversion functions to another file, which is probably a good idea anyway.

KevinWu098 commented on src/lib/components/summary/ScheduledMeetingsList.svelte:
```diff
@@ -0,0 +1,72 @@
+    <div class="flex flex-col gap-2">
+      {#each groupedMeetingsByDate[date] as meeting}
+        <div
+          class="flex flex-col justify-between gap-4 p-3 bg-center bg-cover rounded-lg md:items-center h-fit md:flex-row card hover:variant-ghost"
```
Just going to note here that the card will remain outlined on mobile when using the radio group, but it's far more muted

MinhxNguyen7 reviewed (approved):
Nice work!

```
        _    _
     __|_|__|_|__
   _|____________|__
  |o o o o o o o o / 
~'`~'`~'`~'`~'`~'`~'`~
```
